src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2131,9 +2135,50 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
         _set_remote_or_local_git(cmd, functionapp, resource_group_name, name, deployment_source_url,
                                  deployment_source_branch, deployment_local_git)
 
+    if create_app_insights:
+        try:
+            try_create_application_insights(cmd, functionapp)
+        except Exception: # pylint: disable=broad-except
```

Comment:
Need to catch all kinds of Exception here, because creating an application insights is a safe execution. Had to disable pylint.